nvm.sh

```diff
@@ -2124,6 +2125,51 @@ nvm() {
       return 42
     ;;
 
+    "self-upgrade")
+      local current_nvm_version
+      local latest_nvm_version
+      local origin_nvm_version
+
+      current_nvm_version="$(nvm --version)"
+
+      if nvm_has "git" && [ -d "$NVM_DIR/.git" ]; then
+        command cd "$NVM_DIR"
+        command git fetch --tags
+        latest_nvm_version="$(command git describe --abbrev=0 --tags --match 'v[0-9]*' origin)"
+        if [ "v$current_nvm_version" = "$latest_nvm_version" ]; then
+          nvm_echo "You already have the latest nvm ($latest_nvm_version) "
+        else
+          command git reset --hard
+          command git checkout $latest_nvm_version
+          [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+          origin_nvm_version="$current_nvm_version"
+          current_nvm_version="$(nvm --version)"
+          if [ "v$current_nvm_version" = "$latest_nvm_version" ]; then
+            nvm_echo "nvm successfully upgrade from v$origin_nvm_version to v$current_nvm_version!"
+          else
+            nvm_err  "nvm upgrade failed!!!"
+          fi
+        fi
+      elif nvm_has "nvm_download" && nvm_has "nvm_get_latest"; then
+        latest_nvm_version="$(nvm_get_latest)"
+        if [ "v$current_nvm_version" = "$latest_nvm_version" ]; then
+          nvm_echo "You already have the latest nvm ($latest_nvm_version) "
+        else
+          nvm_download "https://raw.githubusercontent.com/creationix/nvm/$latest_nvm_version/install.sh" | command bash
+          [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
```

Comment:
to silence the shellcheck error here, add `# shellcheck source=/dev/null` as the line preceding it